plugins/graph_algorithm/src/algorithms/components.cpp

```diff
@@ -0,0 +1,73 @@
+#include "graph_algorithm/algorithms/components.h"
+
+#include "graph_algorithm/netlist_graph.h"
+
+#include <map>
+
+namespace hal
+{
+    namespace graph_algorithm
+    {
+        Result<std::vector<std::vector<u32>>> get_connected_components(NetlistGraph* graph, bool strong, u32 min_size)
+        {
+            if (graph == nullptr)
+            {
+                return ERR("graph is a nullptr");
+            }
+
+            igraph_vector_int_t membership, csize;
+            igraph_integer_t number_of_clusters;
+            auto err = igraph_vector_int_init(&membership, 0);
+            if (err != IGRAPH_SUCCESS)
+            {
+                igraph_vector_int_destroy(&membership);
+                return ERR(igraph_strerror(err));
+            }
+
+            err = igraph_vector_int_init(&csize, 0);
+            if (err != IGRAPH_SUCCESS)
+            {
+                igraph_vector_int_destroy(&membership);
+                igraph_vector_int_destroy(&csize);
+                return ERR(igraph_strerror(err));
+            }
+
+            igraph_t* igr = graph->get_graph();
+
+            // run scc
+            err = igraph_clusters(igr, &membership, &csize, &number_of_clusters, strong ? IGRAPH_STRONG : IGRAPH_WEAK);
+            if (err != IGRAPH_SUCCESS)
+            {
+                igraph_vector_int_destroy(&membership);
+                igraph_vector_int_destroy(&csize);
+                return ERR(igraph_strerror(err));
+            }
+
+            // map back to HAL structures
+            u32 num_vertices = (u32)igraph_vcount(igr);
```

Comment:
Nothing wrong with this line, I just wanted to point out that igraph uses `igraph_integer_t` for all integers. This is a _signed_ type. Its size is configurable at build time to be 32- or 64-bit, and by default it follows the word size of the machine.